### ISSUE TITLE
Assertion failed: (stateno <= YY_SHIFT_COUNT), function yy_find_shift_action, file lemon-bug.c, line 512.

### ISSUE THREAD

**Issue by ksherlock** [ksherlock/lemon--#1]
Reported on the sqlite mailing list:
```

%include {
    #include <assert.h>
    #include <stdio.h>
    #include <stdlib.h>
    #include "lemon-bug.h"
}

%code {
    int main()
    {
        void *pParser;

        pParser = ParseAlloc(malloc);
        if (!pParser)
        {
            printf("out of memory\n");
            exit(1);
        }

        ParseTrace(stderr, "Debug: ");

        Parse(pParser, CMDNAME, 0);
        Parse(pParser, INTEGER, 0);
        Parse(pParser, TEXT, 0);
        Parse(pParser, EOL, 0);

        Parse(pParser, CMDNAME, 0);
        Parse(pParser, INTEGER, 0);
        Parse(pParser, TEXT, 0);
        Parse(pParser, EOL, 0);

        Parse(pParser, 0, 0);
        ParseFree(pParser, free);
        return 0;
    }
}


database ::= entrylist.

entrylist ::= command.
entrylist ::= entrylist command.

command ::= CMDNAME cmdargs EOL. 
cmdargs ::= .
cmdargs ::= cmdargs cmdarg.

cmdarg ::= INTEGER.
cmdarg ::= TEXT.

```

**Comment by ksherlock** [comment]
```
./lemon-bug 
Debug: Input 'CMDNAME'
Debug: Shift 'CMDNAME', go to state 3
Debug: Return. Stack=[CMDNAME]
Debug: Input 'INTEGER'
Assertion failed: (stateno <= YY_SHIFT_COUNT), function yy_find_shift_action, file lemon-bug.c, line 512.
Abort trap: 6
```

**Comment by ksherlock** [comment]
`-c` to prevent table compression prevents the issue.

**Comment by ksherlock** [comment]
fixed via b11e73db1d8eac8240d3d9ef1c19b0a15fe47a13 IIRC
